platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorServiceTests.java

```diff
@@ -48,17 +59,32 @@ void exceptionsFromInvalidConfigurationAreNotSwallowed() {
 		assertThat(exception).rootCause().isInstanceOf(IllegalArgumentException.class);
 	}
 
-	@Test
-	@Timeout(5)
-	void defersTasksWithIncompatibleLocks() throws Exception {
-		var configuration = new DefaultParallelExecutionConfiguration(2, 2, 2, 2, 1, __ -> true);
+	static List<Arguments> incompatibleLockCombinations() {
+		return List.of(//
+			arguments(//
+				Set.of(GLOBAL_READ), //
+				Set.of(GLOBAL_READ_WRITE) //
+			), //
+			arguments(//
+				Set.of(new ExclusiveResource("a", LockMode.READ)), //
+				Set.of(new ExclusiveResource("a", LockMode.READ_WRITE)) //
+			), //
```

Comment:
```suggestion
			arguments(//
				Set.of(new ExclusiveResource("a", LockMode.READ)), //
				Set.of(new ExclusiveResource("a", LockMode.READ_WRITE)) //
			), //
			arguments(//
				Set.of(new ExclusiveResource("a", LockMode.READ_WRITE)), //
				Set.of(new ExclusiveResource("a", LockMode.READ_WRITE)) //
			), //
```

Reply:
Why would that be incompatible?

Reply:
As discussed, this would break the exclusiveness guarantee of a test (class) holding a RW lock.